### PR TITLE
sync clock

### LINKED ISSUE #845: Sync clock
- [x] Run `bin/configlet sync -u -e clock --yes --docs --filepaths --metadata --tests include` (updates the Markdown files and maybe `tests.toml`)
  - metadata: unsynced: clock
- [x] Drop strict types comments from test file and example code (these are useless)
- [x] Add and sync test meta data to tests (`uuid` / `@testdox` in DocBlocks)
- [x] Decide on adding / adjusting / ordering test cases to match current problem specs
  - metadata: unsynced: clock

Do not redesign the student's interface or add test cases that would invalidate existing community solutions. These are extra tasks, which should be discussed in advance.

### PR DESCRIPTION
**Pull request by tomasnorre**
Resolves #845

### CODE REVIEW

File: exercises/practice/clock/ClockTest.php [hunk 1]
```diff
@@ -31,204 +9,527 @@ public static function setUpBeforeClass(): void
...
+    /**
+     * uuid a577bacc-106b-496e-9792-b3083ea8705e
+     * @testdox on the hour
+     */
     public function testOnTheHour(): void
     {
         $clock = new Clock(8);
 
-        $this->assertEquals('08:00', $clock->__toString());
+        $this->assertEquals('08:00', $clock);
```
Comment (mk-mxp) [inline exercises/practice/clock/ClockTest.php:20]: This is one of the magic tricks in PHP, but we currently have no concept (concept exercise) for students to learn this. So we should either explain the magic in the stub (for now) or explicitly call the method as done before:

```suggestion
        $this->assertEquals('08:00', $clock->__toString());
```

or

```suggestion
        $this->assertEquals('08:00', (string)$clock);
        // plus use `implements Stringable` in the students stub
        // plus a comment in the stub pointing to PHP docs on interface `Stringable`
```

@homersimpsons What do you think about this?

**Review by mk-mxp — CHANGES_REQUESTED** [review]
Also we should have a comment in the students stub that missing methods need to be added.

File: exercises/practice/clock/ClockTest.php [hunk 2]
```diff
@@ -31,204 +9,527 @@ public static function setUpBeforeClass(): void
...
+    {
+        $clock = new Clock(2, 20);
+        $clock = $clock->sub(3000);
+        $this->assertEquals("00:20", $clock);
     }
 
-    public function testEquivalentClocks(): void
+    /**
+     * uuid f2fdad51-499f-4c9b-a791-b28c9282e311
+     * @testdox clocks with same time
```
Comment (mk-mxp) [inline exercises/practice/clock/ClockTest.php:394]: And the last outer `cases` change:

```suggestion
     * @testdox Compare two clocks for equality - clocks with same time
```

**Review by mk-mxp — CHANGES_REQUESTED** [review]
These changes are not connected to the discussion about the magic method usage.

Please add a comment to the students stub, that all missing methods need to be added by the student.

File: exercises/practice/clock/ClockTest.php (hunk 1 above)
Comment (homersimpsons) [reply exercises/practice/clock/ClockTest.php:20]: If we want to show magic methods with this exercise, I think I would prefer the explicit `(string)` cast, it has the advantages to clearly show that this method is magic.

But I do not know if we want to show those here.

I believe we could have a concept to show `__toString`, `__set`, `__get` which I think are rather "common". `__construct` is a magic method but I believe it is already being taught.

File: exercises/practice/clock/Clock.php [hunk 3]
```diff
@@ -26,6 +26,17 @@
...
+     * This class implements PHP's magic method __toString().
+     *
+     * By implementing this method, the class adheres to the `Stringable` interface.
+     * When an object of this class is used in string context (e.g., echo or string cast),
+     * this method is automatically called.
+     *
+     * More on `Stringable`: https://www.php.net/manual/en/class.stringable.php
+     *
+     * @return string The string representation of the Clock object
+     */
```
Comment (mk-mxp) [inline exercises/practice/clock/Clock.php:39]: Great!

**Review by mk-mxp — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**